lib/gollum-lib/filter/emoji.rb

```diff
@@ -0,0 +1,23 @@
+# ~*~ encoding: utf-8 ~*~
+
+# Emoji
+#
+# Render emoji such as :smile:
+class Gollum::Filter::Emoji < Gollum::Filter
+  EMOJI_TAG_PATTERN = %r{
+    (?<!\[{2})
+    :(?:([\w-]|</?em>)+):
+    (?!\]{^2})
+  }ix
+
+  def extract(data)
+    data
+  end
+
+  def process(data)
+    data.gsub EMOJI_TAG_PATTERN do |tag|
+      name = tag[1..-2].gsub(%r{</?em>}, '_')   # undo emphasis
```

Comment:
Take the emoji `:do_not_litter:`. Since the markdown parser kicks in first, by the time the emoji is parsed, it will be `:do<em>not</em>litter:`. Of course, this collision could be solved properly elsewhere, but since emoji are an opt-in feature, I figured this should be taken care of in the context of the emoji even if it appears a little hackish.